File: movie_handler/src/main/java/com/example/model/Person.java
package com.example.model;

import com.example.searchService.Search;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

import java.util.Date;
import java.util.UUID;

@MappedSuperclass
@Entity
public abstract class Person {
    @Id
    private UUID profileImage;

    private String firstName;
    private String lastName;
    private String nationality;
    private Date dateOfBirth;
    private String biography;

    public Person(String firstName, String lastName, Date dateOfBirth, String nationality, String biography, UUID profileImage) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.dateOfBirth = dateOfBirth;
        this.nationality = nationality;
        this.biography = biography;
        this.profileImage = profileImage;
    }

    public Person() {

    }

    public abstract void accept(Search searchObject);

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Date getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(Date dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public String getNationality() {
        return nationality;
    }

    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public String getBiography() {
        return biography;
    }

    public void setBiography(String biography) {
        this.biography = biography;
    }

    public UUID getProfileImage() {
        return profileImage;
    }

    public void setProfileImage(UUID profileImage) {
        this.profileImage = profileImage;
    }
}

File: recommendation_system/models/person.py
from abc import ABC, abstractmethod
import datetime


class Person(ABC):
    def __init__(self, id: int, first_name: str, last_name: str, date_of_birth: datetime, biography: str, nationality: str):
        self.__id = id
        self.__first_name = first_name
        self.__last_name = last_name
        self.__date_of_birth = date_of_birth
        self.__biography = biography
        self.__nationality = nationality

    def set_first_name(self, first_name: str):
        self.__first_name = first_name

    def get_first_name(self):
        return self.__first_name

    def set_last_name(self, last_name: str):
        self.__last_name = last_name

    def get_last_name(self):
        return self.__last_name

    def set_date_of_birth(self, date_of_birth: datetime):
        self.__date_of_birth = date_of_birth

    def get_date_of_birth(self):
        return self.__date_of_birth

    def set_biography(self, biography: str):
        self.__biography = biography

    def get_biography(self):
        return self.__biography

    def set_nationality(self, nationality: str):
        self.__nationality = nationality

    def get_nationality(self):
        return self.__nationality
